esi_leap/objects/contract.py

```diff
@@ -99,3 +119,22 @@ def expire(self, context=None):
         # expire contract
         self.status = statuses.EXPIRED
         self.save(context)
+
+    @classmethod
+    def split(cls, context, c):
+        # split contract with multiple offers into serval contracts
+        offers = c['offers']
+        contracts = []
+        for o_uuid in offers:
+            contract_dict = {}
+            o = esi_leap.objects.offer.Offer.get(context, o_uuid).to_dict()
+            contract_dict['offer_uuid'] = o['uuid']
+            contract_dict['project_id'] = o['project_id']
+            contract_dict['start_date'] = c['start_time']
+            contract_dict['end_date'] = c['end_time']
+            contract_dict['status'] = c['status']
+            contract_dict['cost'] = o['properties'].get('floor_price', 0)
+            contract_dict['created_at'] = c['time_created']
+            contract_dict['updated_at'] = None
+            contracts.append(contract_dict)
+        return contracts
```

Comment:
I think I'd put this method into something like `common/flocx_market.py`, just because it's not really a method that deals with provider contracts.

Reply:
I've moved this method to common/flocx_market.py. Since the get_contract api in marketplace for provider hasn't implement yet, provider cannot establish a real connection with marketplace. I'm writing test for pulling and pushing contracts to marketplace.